src/Logging.AzureAppServices/src/AzureAppServicesLoggerFactoryExtensions.cs

```diff
@@ -17,6 +17,22 @@ namespace Microsoft.Extensions.Logging
     /// </summary>
     public static class AzureAppServicesLoggerFactoryExtensions
     {
+        internal static string CustomPrefix;
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="customPrefix"></param>
+        /// <returns></returns>
+        public static ILoggingBuilder AddAzureWebAppDiagnostics(this ILoggingBuilder builder, string customPrefix)
+        {
+            CustomPrefix = customPrefix;
+            var context = WebAppContext.Default;
```

Comment:
The custom prefix should be passed into the constructor.

Reply:
@davidfowl 
This is inside a static class, where there is no constructor. 

AddAzureWebAppDiagnostics is an overload extension of ILoggingBuilder, which is the "entry point" to this API. The web application program injects this API to the IHost container like this:

 ihost.ConfigureLogging(x =>
            {
                x.AddAzureWebAppDiagnostics(customPrefixName);
            });

as opposed to the original version, where the DI is coded like this:

 ihost.ConfigureLogging(x =>
            {
                x.AddAzureWebAppDiagnostics();
            });

Notice that all I did was adding that customPrefixName option and store its value in a static field where the provider checks whether to use the custom prefixed flat blob filename logic. I didn't want to touch the original method if possible, out of respect to the original programmer. Please let me know if there is a more preferable approach to implement this logic

Reply:
This is a static class extension entry point to IHostBuilder ConfigureLogging Dependency Injection. It takes the responsibility of a constructor. I'm sure you are familiar with the use of Logging Provider Dependency Injection to IHostBuilder ConfigureLogging. Please revisit this code, I appreciate your time!

Reply:
We don't design anything like this, statics are pretty much forbidden, especially mutable ones like this. This needs to be a constructor argument to the instance.